dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,306 @@
+﻿#include "pch.h"
+#include "NumberBoxParser.h"
+
+// Pass by reference
+MathToken::MathToken(TokenType t, std::wstring& s)
+{
+    this->type = t;
+    this->str = s;
+}
+
+// Pass directly
+MathToken::MathToken(TokenType t, std::wstring s)
+{
+    this->type = t;
+    this->str = s;
+}
+
+MathTokenizer::MathTokenizer(std::wstring_view input)
+{
+    m_inputString = input;
+    m_inputLength = static_cast<int>(m_inputString.size());
+    m_lastToken = MathToken(MathToken::TokenType::EOFToken, L"");
+}
+
+MathToken MathTokenizer::GetToken(ExpressionType type)
+{
+    // Skip Whitespaces
+    SkipWhiteSpace();
+
+    if (m_index >= m_inputLength)
+    {
+        return MathToken(MathToken::TokenType::EOFToken, L"");
+    }
+ 
+    std::wstringstream ss;
+    ss << m_inputString[m_index];
+
+    // Check for a negative number. Next Token must be numeric and previous token must not be numeric.
```

Comment:
Couldn't the next token be a '('  ?

Reply:
As in "2 * -(3 + 2)"? It doesn't work currently. Added to #1505.